website/content/docs/nia/configuration.mdx

```diff
@@ -137,6 +140,67 @@ task {
+- `condition` - (obj) The requirement that, when met, triggers Consul-Terraform-Sync to execute the task. When unconfigured, the default condition is to trigger the task on changes in the services configured in [`services`](#services). Only one `condition` may be configured per task. Consul-Terraform-Sync supports different types of conditions, which each have their own configuration options. See [Task Condition](#task-condition) configuration for full details on configuration options for each condition type.
+
+### Task Condition
+
+A `task` block can be optionally configured with a `condition` block to set the conditions that should be met in order to execute that particular task. Below are the different types of conditions that Consul-Terraform-Sync supports.
+
+#### Services Condition
+
+This is the default type of condition when no `condition` block is configured for a task. This condition will trigger the task on changes in services configured in [`task.services`](#services). The task can be optionally configured with a services condition block. However, this is not necessary since this is the default type of condition and there is currently no additional configuration options available for services condition.
+
+See [Task Execution: Services Condition](/docs/nia/tasks#services-condition) for more details on how tasks are triggered with a services condition.
+
+```hcl
+task {
+  name = "services_condition_task"
+  description = "execute on changes to api, db, and web services"
+  services = ["api", "db", "web"]
+  providers = ["my-provider"]
+  source = "path/to/services-condition-module"
+
+  // optional: declare empty services condition block
+  condition "services" {
+    // no additional configuration options currently
+  }
+}
+```
+
+#### Catalog-Services Condition
+
+A catalog-services condition block configures a task to only execute on service registration and deregistration, more specifically on first service instance registration and last service instance deregistration respectively. The catalog-services condition has additional configuration options to specify the services that can trigger the task on registration and deregistration.
+
+See [Task Execution: Catalog Services Condition](/docs/nia/tasks#catalog-services-condition) for more information on how tasks are triggered with a catalog-services condition.
+
+```hcl
+task {
+  name = "catalog_service_condition_task"
+  description = "execute on service de/registrations with name matching 'web.*'"
+  source = "path/to/catalog-service-module"
+  providers = ["my-provider"]
+
+  // configure depending on module. provides additional services information but
+  // does not execute task. refer to module docs on how to configure.
+  services = ["web-api"]
+
+  condition "catalog-service" {
```

Comment:
`service` block allows a user to input information related to datacenter and namespace. In a case where the user has a `service` block, in this example "web-api" with namespace = "foo" and datacenter = "bar" and defines a task with services = ["web-api"] and condition "catalog-service" with datacenter = "dc2" and namespace = "namespace2", what would be the behavior? 
It would be useful to identify the behavior

Reply:
Good point! I've only mentioned this abstractly. I added an explicit example on the Tasks page ([preview](https://consul-ahn02uelx-hashicorp.vercel.app/docs/nia/tasks#catalog-services-condition)) as well and added a link to it from here (the configuration page). Let me know if you have any thoughts! https://github.com/hashicorp/consul/pull/10376/commits/672efbade9f5c47f5fab2878b55833048d1e1714